runtime/major_gc.c

```diff
@@ -700,7 +700,9 @@ Caml_noinline static intnat do_some_marking(intnat work)
 
     for (; scan < scan_end; scan++) {
       value v = *scan;
+      CAML_EVENTLOG_DO({ (*slice_fields) ++; });
       if (Is_major_block(v)) {
+        CAML_EVENTLOG_DO({ (*slice_pointers) ++; });
```

Comment:
Here, gcc -O2 is very bad at optimising this code. This is in the hot loop and will affect the thing being measured in the instrumented runtime. How about something like
```c
#ifdef CAML_INSTR
  pointers++;
#endif
```
?

Reply:
Good idea, I hadn't realised how much overhead CAML_EVENTLOG_DO would introduce.